### DISCUSSION TITLE
Make discounts, sales, shipping mothods etc into elements

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2424] [Ideas]
It would be auseful if we could attach additional information to shipping and payment methods using field layouts.

This would be useful if we wanted to add logo to be displayed on shipping selection form or maybe some longer description.

**Comment by lukeholder** [comment]
Renaming this request to be a general 'Make more models elements'.

- Discounts
- Sales
- Shipping Methods
- Shipping Rules
- Tax Rates

Allowing all of the above to have custom fields makes sense.

**Comment by Anubarak** [comment]
I would totally love this. Currently I hacked custom field layouts into these and it's neither clean nor do I like the approach I had to do.

**Comment by jelmer-scoremedia** [comment]
We really need this too, to add custom logic on top of discounts.
